src/gamestate.rs

```diff
@@ -1,795 +0,0 @@
-            moves_since: 0,
-        };
-        game_state.intialise_new_board();
-        game_state
-    }
-    pub fn slected_piece_coord(&self) -> Option<Point> {
-        self.selected.as_ref().map(|s| s.starting_tile)
-    }
-
-    pub fn is_empty(&self, (x, y): Point) -> bool {
-        Tile::Empty == self.board[x as usize][y as usize]
-    }
-
-    pub fn mouse_down(&mut self, x: i32, y: i32) {
-        let (board_x, board_y) = get_board_position((x, y));
-        if let Some(ref selected) = self.selected {
-            if self.is_valid_tile(board_x, board_y) {
-                self.move_piece(Move::new(selected.starting_tile, (board_x, board_y)));
-                return;
-            }
-        }
-        self.select_tile((x, y));
-    }
-
-    pub fn mouse_up(&mut self, x: i32, y: i32) {
-        let (board_x, board_y) = get_board_position((x, y));
-        if let Some(
-            ref selected @ Selection {
-                held_down: true, ..
-            },
-        ) = self.selected
-        {
-            if self.is_valid_tile(board_x, board_y) {
-                self.move_piece(Move::new(selected.starting_tile, (board_x, board_y)));
-            } else {
-                self.selected
-                    .as_mut()
-                    .expect("checked in if statement")
-                    .held_down = false;
-            }
-        }
-    }
-
-    pub fn mouse_move(&mut self, x: i32, y: i32) {
-        if let Some(ref mut selected) = self.selected {
-            selected.current_point = (x, y);
-        }
-    }
-
-    pub fn legal_moves(&self) -> HashSet<Point> {
-        if let Some(ref selected) = self.selected {
-            let selected_tile = MovingPiece::new(selected.piece, selected.starting_tile);
-            self.valid_piece_moves(&selected_tile)
-        } else {
-            HashSet::new()
-        }
-    }
-
-    pub fn select_tile(&mut self, (pos_x, pos_y): (i32, i32)) {
-        let (board_x, board_y) = get_board_position((pos_x, pos_y));
-        if !in_bounds((board_x, board_y)) {
-            return;
-        }
-        if let Tile::Piece(piece) = self.board[board_x as usize][board_y as usize] {
-            if piece.color == self.players_turn {
-                self.selected = Some(Selection {
-                    starting_tile: (board_x, board_y),
-                    current_point: (pos_x, pos_y),
-                    piece,
-                    held_down: true,
-                });
-            }
-        }
-    }
-
-    pub fn get_moving_piece(&self) -> Option<(Piece, Point)> {
-        if let Some(Selection {
-            held_down,
-            piece,
-            current_point,
-            ..
-        }) = self.selected
-        {
-            if held_down {
-                return Some((piece, current_point));
-            }
-        }
-        None
-    }
-
-    pub fn intialise_new_board(&mut self) {
-        self.board = vec![vec![Tile::Empty; BOARD_LENGTH as usize]; BOARD_LENGTH as usize];
```

Comment:
if the size never changes why not use an array?

Reply:
i remember using a array before when i was making othello and i had issue i can't remember what they were